lib/active_model/serializer.rb

```diff
@@ -124,17 +127,37 @@ def self.root_name
       name.demodulize.underscore.sub(/_serializer$/, '') if name
     end
 
+    def self.digest_caller_file(caller_line)
+      serializer_file_path = caller_line[CALLER_FILE]
+      serializer_file_contents = IO.read(serializer_file_path)
+      Digest::MD5.hexdigest(serializer_file_contents)
+    end
+
+    def self.get_serializer_for(klass)
+      serializers_cache.fetch_or_store(klass) do
+        serializer_class_name = "#{klass.name}Serializer"
+        serializer_class = serializer_class_name.safe_constantize
+
+        if serializer_class
+          serializer_class
+        elsif klass.superclass
+          get_serializer_for(klass.superclass)
+        end
+      end
+    end
+
     attr_accessor :object, :root, :meta, :meta_key, :scope
+    attr_reader :options
 
-    def initialize(object, options = {})
-      @object = object
-      @options = options
-      @root = options[:root]
-      @meta = options[:meta]
-      @meta_key = options[:meta_key]
-      @scope = options[:scope]
+    def initialize(object, opts = {})
```

Comment:
Good eye.  Because there is an `options` method set by `@options` method and `options` arguments passed to method that are not the same thing, so I thought to distinguish.  Methods blow up more more nicely than ivars, too :)